contracts/TridentRouter.sol

```diff
@@ -220,14 +221,15 @@ contract TridentRouter is ITridentRouter, SelfPermit, Multicall {
     /// @notice Recover mistakenly sent tokens.
     function sweep(
         address token,
-        uint256 amount,
         address recipient,
         bool fromBento
     ) external payable {
         if (fromBento) {
-            bento.transfer(token, address(this), recipient, amount);
+            uint256 shares = bento.balanceOf(token, address(this));
+            bento.transfer(token, address(this), recipient, shares);
         } else {
-            token == USE_NATIVE ? recipient.safeTransferETH(address(this).balance) : token.safeTransfer(recipient, amount);
+            uint256 amount = (token == USE_NATIVE ? address(this).balance : IERC20(token).balanceOf(address(this))) - 1;
```

Comment:
@sarangparikh22 does it make sense to leave 1 unit of ETH in the router too? Unsure if this saves gas in the same way? Also, unwrap ETH will always empty it anyway.

Reply:
@matthewlilley Nice catch! Updated!